### PR TITLE
Metadata enricher

### PR DESCRIPTION
**Pull request by MattKetmo**
Supersede #157

### CODE REVIEW

File: src/Metadata/MetadataEnricherAggregate.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+namespace Prooph\EventStore\Metadata;
+
+use Prooph\Common\Messaging\Message;
+
+final class MetadataEnricherAggregate implements MetadataEnricher
+{
+    /**
+     * @var array
```
Comment (codeliner) [inline src/Metadata/MetadataEnricherAggregate.php]: type hint MetadataEnricher[]

File: src/Metadata/MetadataEnricherAggregate.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    private $metadataEnrichers;
+
+    /**
+     * @param array $metadataEnrichers
```
Comment (codeliner) [inline src/Metadata/MetadataEnricherAggregate.php]: type hint MetadataEnricher[]

File: src/Metadata/MetadataEnricherAggregate.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    public function __construct(array $metadataEnrichers)
+    {
+        $this->metadataEnrichers = $metadataEnrichers;
```
Comment (codeliner) [inline src/Metadata/MetadataEnricherAggregate.php]: Assert::allInstanceOf(...)

**Comment by codeliner** [comment]
@MattKetmo Looks great! Just two minor remarks and we should add a note in the docs. I'd say a short section at the end of the [event_store.md](https://github.com/prooph/event-store/blob/master/docs/event_store.md) file is enough. We've defined two ways for adding docs, either with the PR or by creating a doc issue and link it with the PR. Goal is to have both ready before the feature is merged into master.

**Comment by MattKetmo** [comment]
Type hint fixed + "*Metadata enricher*" section [added to the doc](https://github.com/MattKetmo/event-store/blob/metadata-enricher/docs/event_store.md#metadata-enricher) with an example of usage.

I will later submit another PR to integrate this with the ContainerInterop as suggested in previous PR. Something like:

```php
return [
    'prooph' => [
        'event_store' => [
            'metadata_enrichers' => [
                UserMetadataEnricher::class,
                OtherMetadataEnricher::class,
            ],
    // ...
```

**Pull request merged**